netdev.go

```diff
@@ -55,14 +55,86 @@ type netdever interface {
 	// DHCP or statically
 	Addr() (netip.Addr, error)
 
-	// Berkely Sockets-like interface, Go-ified.  See man page for socket(2), etc.
-	Socket(domain int, stype int, protocol int) (int, error)
+	// # Socket Address family/domain argument
+	//
+	// Socket address families specifies a communication domain:
+	//  - AF_UNIX, AF_LOCAL(synonyms): Local communication For further information, see unix(7).
+	//  - AF_INET: IPv4 Internet protocols.  For further information, see ip(7).
+	//
+	// # Socket type argument
+	//
+	// Socket types which specifies the communication semantics.
+	//  - SOCK_STREAM: Provides sequenced, reliable, two-way, connection-based
+	//  byte streams.  An out-of-band data transmission mechanism may be supported.
+	//  - SOCK_DGRAM: Supports datagrams (connectionless, unreliable messages of
+	//  a fixed maximum length).
+	//
+	// The type argument serves a second purpose: in addition to specifying a
+	// socket type, it may include the bitwise OR of any of the following values,
+	// to modify the behavior of socket():
+	//  - SOCK_NONBLOCK: Set the O_NONBLOCK file status flag on the open file description.
+	//
+	// # Socket protocol argument
+	//
+	// The protocol specifies a particular protocol to be used with the
+	// socket.  Normally only a single protocol exists to support a
+	// particular socket type within a given protocol family, in which
+	// case protocol can be specified as 0. However, it is possible
+	// that many protocols may exist, in which case a particular
+	// protocol must be specified in this manner.
+	//
+	// # Return value
+	//
+	// On success, a file descriptor for the new socket is returned. Quoting man pages:
+	// "On error, -1 is returned, and errno is set to indicate the error." Since
```

Comment:
This discussion on errno/error is good and applies to all the socket funcs that return error.  Can we move this up higher in the file as a note to apply to all socket funcs?